front/src/modules/pathfinding/components/IncompatibleConstraints/IncompatibleConstraintsLayer.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { type FC } from 'react';
+
+import type { FeatureCollection, LineString } from 'geojson';
+import type { FilterSpecification } from 'maplibre-gl';
+import { Layer, Source } from 'react-map-gl/maplibre';
+
+function buildLayoutFilter(
+  ids: Set<string>,
+  predicate: 'all' | 'any'
+): FilterSpecification | undefined {
+  return [
+    predicate,
+    ...Array.from(ids).map((id) => ['in', id, ['get', 'ids']]),
+  ] as FilterSpecification;
+}
+interface IncompatibleConstraintsLayerProps {
+  data: FeatureCollection<LineString, { ids: string[] }>;
+  selected: Set<string>;
+  hovered: Set<string>;
```

Comment:
To improve readability ?

```suggestion
  filteredConstraintsGeojson: FeatureCollection<LineString, { ids: string[] }>;
  selectedConstraintsIds: Set<string>;
  hoveredConstraintsIds: Set<string>;
```